examples/advanced/examples/reth_db_provider.rs

```diff
@@ -0,0 +1,237 @@
+//! `ProviderCall` enables the alloy-provider to fetch results of a rpc request from arbitrary
+//! sources. These arbitray sources could be a RPC call over the network, a local database, or even
+//! a synchronous function call.
+//!
+//! `ProviderCall` is the final future in the flow of an rpc request and is used by the
+//! `RpcWithBlock` and `EthCall` types under the hood to give flexibility to the user to use
+//! their own implementation of the `Provider` trait and fetch results from any source.
```

Comment:
can we rephrase this so that this describes first what this example does and then focus on how it does it (ProviderCall)